Wikipedia/Code/WMFExploreViewController.m

```diff
@@ -157,6 +163,18 @@ - (BOOL)rowAtIndexPathIsOnlyRowVisibleInSection:(NSIndexPath*)indexPath {
     }
 }
 
+/**
+ *  Check whether or not a section is going in or out of view.
+ *
+ *  @param indexPath The index path of the row which will or did end displaying.
+ *
+ *  @return @c YES if that section isn't displaying or the given row is the only one visible in its section, otherwise @c NO.
+ */
+- (BOOL)isVisibilityTransitioningForRowIndexPath:(NSIndexPath*)indexPath {
+    return ![self isDisplayingCellsForSection:indexPath.section]
```

Comment:
this method encapsulates the logic we're using in `didEndDisplayingCell` (and previously in `willDisplayCell`). the idea is that the visibility is "transitioning" if either:

1. the section isn't visible (yet, either scrolling on/off screen)
2. the specified row is the only one visible (first row to appear)